src/containers/SecretsModal/SecretsModal.js

```diff
@@ -0,0 +1,273 @@
+/*
+Copyright 2019 The Tekton Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import { Modal } from 'carbon-components-react';
+import UniversalFields from '../../components/SecretsModal/UniversalFields';
+import Annotations from '../../components/SecretsModal/Annotations';
+import BasicAuthFields from '../../components/SecretsModal/BasicAuthFields';
+import '../../components/SecretsModal/SecretsModal.scss';
+/* istanbul ignore next */
+import { createSecret } from '../../actions/secrets';
+import { fetchNamespaces } from '../../actions/namespaces';
+import { getNamespaces } from '../../reducers';
+
+/* istanbul ignore next */
+function validateK8sResources(value, id) {
+  if (value.trim() === '') return false;
+
+  const trimmed = value.trim();
+
+  if (id === 'name' || id === 'serviceAccount') {
+    if (trimmed.length > 253) return false;
+
+    if (/[^-.a-z1-9]/.test(trimmed)) return false;
+  }
+
+  return true;
+}
+
+const initialState = {
+  name: '',
+  namespace: '',
+  accessTo: '',
+  username: '',
+  password: '',
+  serviceAccount: '',
+  annotations: [`tekton.dev/git-0`, ''],
+  invalidFields: []
+};
+
+export /* istanbul ignore next */ class SecretsModal extends Component {
+  constructor(props) {
+    super(props);
+    this.state = initialState;
+  }
+
+  componentDidMount() {
+    this.props.fetchNamespaces();
```

Comment:
shouldn't be necessary, namespaces are loaded globally by the nav and kept up to date via websocket events